tests/acceptance/searchrequestlogger.js

```diff
@@ -0,0 +1,66 @@
+import { VERTICAL_SEARCH_URL_REGEX, UNIVERSAL_SEARCH_URL_REGEX } from './constants';
+import { Selector, RequestLogger, t } from 'testcafe';
+import { registerIE11NoCacheHook } from './utils';
+
+/**
+ * Handles search request logger registration and request/response data received during test execution.
+ */
+class SearchRequestLogger {
+  constructor () {
+    this._searchComplete = Selector('.yxt-Results--searchComplete');
+  }
+
+  /**
+   * Register a RequestLogger that tracks vertical query requests to given test.
+   * If browser is IE11, register an Ie11NoCacheHook.
+   *
+   * @param {import('testcafe').TestController} testInstance
+   */
+  async registerVerticalSearchLogger (testInstance) {
+    this._queryRequestLogger = RequestLogger({
+      url: VERTICAL_SEARCH_URL_REGEX
+    });
+    await testInstance.addRequestHooks(this._queryRequestLogger);
+    await registerIE11NoCacheHook(testInstance, VERTICAL_SEARCH_URL_REGEX);
+  }
+
+  /**
+   * Register a RequestLogger that tracks universal query requests to given test.
+   * If browser is IE11, register an Ie11NoCacheHook.
+   *
+   * @param {import('testcafe').TestController} testInstance
+   */
+  async registerUniversalSearchLogger (testInstance) {
+    this._queryRequestLogger = RequestLogger({
+      url: UNIVERSAL_SEARCH_URL_REGEX
+    });
+    await testInstance.addRequestHooks(this._queryRequestLogger);
+    await registerIE11NoCacheHook(testInstance, UNIVERSAL_SEARCH_URL_REGEX);
+  }
+
+  /**
+   * Wait for results to load on page by checking query response status and searchComplete state
+   * (timeout is set to 10 seconds)
+   */
+  async waitOnSearchComplete () {
+    const responseWaitTimeout = 10000;
+    const waitTimeInterval = 200;
+    let totalWaitTime = 0;
+    while (totalWaitTime < responseWaitTimeout && !this.isLoggerResultsPresent()) {
+      await t.wait(waitTimeInterval);
+      totalWaitTime += waitTimeInterval;
+    }
+    await t.expect(this._searchComplete.exists).ok();
```

Comment:
sorry if somebody already asked but do we need to wait for the search complete state? I'm sure at worst it doesn't hurt to have it but I feel like we might not need it

Reply:
`exists` property is evaluated immediately regardless of selector timeout since we wouldn't be waiting in this assert statement. Yeah the results response check is properly sufficient, this is to confirm that the UI setup on page is completely loaded.

Reply:
hmm it would just be an extra check so it's not needed, will remove it!